File: fix-cors.js
const allowedOrigins = process.env.ALLOWED_ORIGINS
  ? process.env.ALLOWED_ORIGINS.split(',').map(o => o.trim())
  : ['https://yourdomain.com', 'http://localhost:3000'];

const corsMiddleware = (req, res, next) => {
  const origin = req.headers.origin;
  if (origin && allowedOrigins.includes(origin)) {
    res.setHeader('Access-Control-Allow-Origin', origin);
    res.setHeader('Access-Control-Allow-Credentials', 'true');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
  }
  // Handle preflight
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }
  next();
};

module.exports = corsMiddleware;

File: fix_cors.py
# fix_cors.py
from flask import Flask, request, jsonify

app = Flask(__name__)

# Lista blanca de orígenes permitidos
ALLOWED_ORIGINS = [
    'https://tu-dominio-seguro.com',
    'https://www.tu-dominio-seguro.com',
    # Agrega aquí todos los dominios que quieras permitir
]

@app.after_request
def add_cors_headers(response):
    """
    Configura los headers CORS de manera segura.
    Solo permite orígenes en la lista blanca.
    """
    origin = request.headers.get('Origin')
    if origin in ALLOWED_ORIGINS:
        response.headers['Access-Control-Allow-Origin'] = origin
        response.headers['Access-Control-Allow-Credentials'] = 'true'
        response.headers['Access-Control-Allow-Methods'] = 'GET, POST, PUT, DELETE, OPTIONS'
        response.headers['Access-Control-Allow-Headers'] = 'Content-Type, Authorization'
    return response

# Ejemplo de uso
@app.route('/api/data', methods=['GET'])
def get_data():
    return jsonify({"message": "Datos seguros con CORS configurado correctamente"})

if __name__ == '__main__':
    app.run(debug=True)